### ISSUE TITLE
FlexAttention results do not match FlashAttention results

### ISSUE THREAD

**Issue by tilmto** [pytorch-labs/attention-gym#50]
Hi,

I noticed that for certain sequence lengths, given the same inputs, FlexAttention's output differs from FlashAttention's output.

For example, with an input sequence of length 137, FlexAttention with a causal mask will, by default, create a block mask of length 256. In this case, when inputting the same set of Q, K, and V to both FlexAttention and FlashAttention, the first 128 tokens of their output will be identical, but the remaining 9 tokens will show slight differences.

I’m wondering if this discrepancy is due to some specific handling of FlexAttention's attention mask, and how we can address it. Thanks!

**Comment by drisspg** [comment]
Do you a repro? Since they are two different implementations with potentially different reduction orders it is expected that there will be some slight numerical differences within the bounds specified here: https://pytorch.org/docs/stable/notes/numerical_accuracy.html

No BlockM and BlockN aligned shapes do need special care (padding and masking locally) to handle which is the case that you run into with sequence length = 137

**Comment by drisspg** [comment]
Do you think you could provide a runnable example, we did have some early bugs when we initially added non multiple of 128 sequence length support, however they should be resolved on nightly.

**Comment by drisspg** [comment]
This should be fixed now
